sqrl-engines/sqrl-engine-jdbc/src/main/java/com/datasqrl/engine/database/relational/JDBCEngine.java

```diff
@@ -161,16 +162,18 @@ private List<SqlDDLStatement> extractTypeExtensions(RelNode relNode) {
     Set<SqlDDLStatement> statements = new HashSet<>();
     //look at relnodes to see if we use a vector type
     for (RelDataTypeField field : relNode.getRowType().getFieldList()) {
-      if (field.getType() instanceof VectorType) {
+      if (field.getType() instanceof RawRelDataType &&
+        ((RawRelDataType) field.getType()).getRawType().getOriginatingClass() == FlinkVectorType.class)
         statements.add(new PostgresCreateVectorExtensionStatement());
-      }
     }
 
+    Set<String> vecFncs = Set.of("cosinesimilarity", "cosinedistance", "euclideandistance",
```

Comment:
Doing that by function name is very brittle imho. Instead, check the operand types generically. You can generalize this with the code above by pulling out all "non-standard" types first into a set and then add any extensions that a dialect might require.